packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx

```diff
@@ -0,0 +1,220 @@
+    };
+  }, []);
+
+  // TODO: remove sortModel check once row reorder is sorting compatible
+  // remove treeDepth once row reorder is tree compatible
+  const isRowReorderDisabled = React.useCallback((): boolean => {
+    return props.disableRowReorder || !!sortModel.length || treeDepth !== 1;
+  }, [props.disableRowReorder, sortModel, treeDepth]);
+
+  const handleDragStart = React.useCallback<GridEventListener<GridEvents.rowDragStart>>(
+    (params, event) => {
+      if (isRowReorderDisabled()) {
+        return;
+      }
+
+      logger.debug(`Start dragging row ${params.id}`);
+      // Prevent drag events propagation.
+      // For more information check here https://github.com/mui/mui-x/issues/2680.
+      event.stopPropagation();
+
+      dragRowNode.current = event.currentTarget;
+      dragRowNode.current.classList.add(classes.rowDragging);
+
+      apiRef.current.setState((state) => ({
+        ...state,
+        rowReorder: { ...state.rowReorder, dragRow: params.id },
+      }));
+      apiRef.current.forceUpdate();
+
+      removeDnDStylesTimeout.current = setTimeout(() => {
+        dragRowNode.current!.classList.remove(classes.rowDragging);
+      });
+
+      originRowIndex.current = apiRef.current.getRowIndex(params.id);
+    },
+    [isRowReorderDisabled, classes.rowDragging, logger, apiRef],
+  );
+
+  const handleDragEnter = React.useCallback<
+    GridEventListener<GridEvents.cellDragEnter | GridEvents.rowDragEnter>
+  >((params, event) => {
+    event.preventDefault();
+    // Prevent drag events propagation.
+    // For more information check here https://github.com/mui/mui-x/issues/2680.
+    event.stopPropagation();
+  }, []);
+
+  const handleDragOver = React.useCallback<
+    GridEventListener<GridEvents.cellDragOver | GridEvents.rowDragOver>
+  >(
+    (params, event) => {
+      const dragRowId = gridRowReorderDragRowSelector(apiRef);
+      if (!dragRowId && dragRowId !== 0) {
+        return;
+      }
+      logger.debug(`Dragging over row ${params.id}`);
+      event.preventDefault();
+      // Prevent drag events propagation.
+      // For more information check here https://github.com/mui/mui-x/issues/2680.
+      event.stopPropagation();
+
+      const coordinates = { x: event.clientX, y: event.clientY };
+
+      if (
+        params.id !== dragRowId &&
+        hasCursorPositionChanged(cursorPosition.current, coordinates)
+      ) {
+        const targetRowIndex = apiRef.current.getRowIndex(params.id);
+        const dragRowIndex = apiRef.current.getRowIndex(dragRowId);
+
+        const cursorMoveDirectionY = getCursorMoveDirectionY(cursorPosition.current, coordinates);
+        const hasMovedTop =
+          cursorMoveDirectionY === CURSOR_MOVE_DIRECTION_TOP && dragRowIndex > targetRowIndex;
+        const hasMovedBottom =
+          cursorMoveDirectionY === CURSOR_MOVE_DIRECTION_BOTTOM && dragRowIndex < targetRowIndex;
+
+        if (hasMovedTop || hasMovedBottom) {
```

Comment:
I know this was copied from the column reorder but is there a need for these checks here? We already know `params.id !== dragRowId`.